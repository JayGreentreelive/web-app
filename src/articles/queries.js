import gql from 'graphql-tag'

export const GET_ALL_ARTICLES = gql`
    query {
        getArticles {
            id
            title

            summary
        
            images {
                sources {
                    uri
                }
            }
        }
    }
`

export const GET_ARTICLE_BY_TITLE = gql`
    query getArticleByTitle($title:String!) {
        getArticleByTitle(title:$title) {
            id
            title
            
            htmlContent
            summary
            readTime
            
            images {
                sources {
                    uri
                }
            }

            author {
                firstName
                lastName
                
                photo {
                    uri
                }
            }

            publishDate
        }
    }
`

export const GET_ARTICLE_CATEGORIES = gql`
    query getArticleCategories($id:ID!) {
        node(id:$id) {
            ... on ArticleContentItem {
                categories
            }
        }
    }
`

export const GET_RELATED_ARTICLES = gql`
    query getRelatedArticles($id:ID!) {
        node(id:$id) {
            ... on ArticleContentItem {
                siblingContentItemsConnection(first:3) {
                    edges {
                      node {
                        title
    
                        images {
                            sources {
                                uri
                            }
                        }
                      }
                    }
                }
            }
        }
    }
`
export const GET_TOP_THREE_ARTICLES = gql`
    query {
        getArticles(first:3) {
            title
        
            images {
                sources {
                    uri
                }
            }
        }
    }
`