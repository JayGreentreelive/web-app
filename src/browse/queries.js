import gql from 'gql-tag'

export const GET_BROWSE_FILTERS = gql`
    query getBrowseFilters {
        contentChannels {
            childContentItemsConnection {
                edges {
                    node {
                        id
                        title
                    }
                }
            }
        }
    }
`

export const GET_CATEGORIES_FROM_FILTER = gql`
    query getCategoriesFromFilter($filterId: ID!) {
        node(id: $filterId) {
        ...on ContentItem {
                title

                childContentItemsConnection {
                    edges {
                        node {
                            id
                            title

                            childContentItemsConnection(first: 3) {
                                edges {
                                    node {
                                        title
                                    }
                                }
                            }

                        }
                    }
                }

            }
        }
    }
`

export const GET_ALL_CONTENT_FROM_CATEGORY = gql`
    query getAllContentFromCategoryId($categoryId: ID!) {
        node(id: $categoryId) {
        ...on ContentItem {
                title
                childContentItemsConnection {
                    edges {
                        node {
                            title
                        }
                    }
                }
            }
        }
    }
`