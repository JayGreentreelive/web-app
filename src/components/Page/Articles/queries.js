import gql from 'graphql-tag'

export const GET_ALL_ARTICLES = gql`
query {
    getArticles {
        id
        title

        summary
    
        images {
            sources {
                uri
            }
        }
    }
}
`

export const GET_ARTICLE_BY_TITLE = gql`
query getArticleByTitle($title:String!) {
    getArticleByTitle(title:$title) {
        id
        title
        
        htmlContent
        summary
        readTime
        
        images {
            sources {
                uri
            }
        }

        author {
            firstName
            lastName
            
            photo {
                uri
            }
        }

    }
}
`

