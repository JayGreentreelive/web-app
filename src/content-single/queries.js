import gql from 'graphql-tag'

export const PUBLISH_FRAGMENT = gql`
    fragment publishFragment on ContentItem {
        ... on ContentSeriesContentItem {
            author {
                firstName
                lastName
                
                photo {
                    uri
                }
            }
            estimatedTime
            publishDate
        }
        ... on UniversalContentItem {
            author {
                firstName
                lastName
                
                photo {
                    uri
                }
            }
            estimatedTime
            publishDate
        }
        ... on DevotionalContentItem {
            author {
                firstName
                lastName
                
                photo {
                    uri
                }
            }
            estimatedTime
            publishDate
        }
        ... on MediaContentItem {
            author {
                firstName
                lastName
                
                photo {
                    uri
                }
            }
            estimatedTime
            publishDate
        }
    }
`

export const ACCESSORY_FRAGMENT = gql`
    fragment accessoryFragment on ContentItem {
        ... on ContentSeriesContentItem {
            tags
            icon
        }
        ... on UniversalContentItem {
            tags
            icon
        }
        ... on DevotionalContentItem {
            tags
            icon
        }
        ... on MediaContentItem {
            tags
            icon
        }
    }
`

export const GET_ARTICLE_BY_TITLE = gql`
    query getContentItemByTitle($title:String!) {
        getContentItemByTitle(title:$title) {
            id
            title
        
            htmlContent
            summary
        
            images {
                sources {
                    uri
                }
            }
        
            videos {
                sources {
                    uri
                }
            }
            ...accessoryFragment
            ...publishFragment
        }
    }
    ${ACCESSORY_FRAGMENT}
    ${PUBLISH_FRAGMENT}
`

export const GET_ARTICLE_CATEGORIES = gql`
    query getArticleCategories($id:ID!) {
        node(id:$id) {
            ... on ContentItem {
                ...accessoryFragment
            }
        }
    }
    ${ACCESSORY_FRAGMENT}
`

export const GET_RELATED_ARTICLES = gql`
    query getRelatedArticles($id:ID!) {
        node(id:$id) {
            ... on ContentItem {
                siblingContentItemsConnection(first:4) {
                    edges {
                      node {
                        id
                        title
    
                        images {
                            sources {
                                uri
                            }
                        }
                      }
                    }
                }
            }
        }
    }
`
export const GET_TOP_THREE_ARTICLES = gql`
    query getBrowseFilters {
        getBrowseFilters {
            id
            childContentItemsConnection(first:1) {
                edges {
                    node {
                        id
                    
                        childContentItemsConnection(first:1) {
                            edges {
                                node {
                                    id
                                
                                    childContentItemsConnection(first:4) {
                                        edges {
                                            node {
                                                id
                                                title
                                            
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`