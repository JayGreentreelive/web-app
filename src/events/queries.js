import gql from 'graphql-tag';

export const GET_EVENT_SCHEDULES = gql`
    query getEventSchedules($id: ID!) {
        node(id: $id) {
            ... on ContentItem {
                id
                childContentItemsConnection {
                    edges {
                        node {
                            id
                            ... on EventScheduleItem {
                                dates
                                campuses {
                                    name
                                }
                                location
                            }
                        }
                    }
                }
            }
        }
    }
`;

export const GET_CURRENT_PERSON_CAMPUS = gql`
    query {
        currentUser {
            profile {
                campus {
                    name
                }
            }
        }
    }
`;

export const GET_CAMPUS_ADDRESSES = gql`
    query getCampuses {
        campuses {
            id
            name
        
            street1
            postalCode
            state
            city
        }
    }
`;