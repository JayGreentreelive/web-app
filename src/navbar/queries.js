import gql from 'graphql-tag'

export const GET_WEBSITE_HEADER = gql`
  query websiteNavigation($website:String!) {
    getWebsiteNavigation(website:$website) {
      id
      
      images {
        sources {
          uri
        }
        name
      }
      
      navigationLinks {
        call
        action
      }
      quickAction {
        call
        action
      }
    }
  }
`;
