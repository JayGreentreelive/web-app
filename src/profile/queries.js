import gql from 'graphql-tag'

export const GET_CAMPUSES = gql`
  query {
    campuses {
      id
      name
    }
  }
`

export const GET_ETHNICITY_LIST = gql`
  query {
    getEthnicityList {
      id
      values {
        id
        value
      }
    }
  }
`

export const GET_STATES_LIST = gql`
  query {
    getStatesList {
      id
      values {
        id
        value
      }
    }
  }
`

export const GET_CURRENT_PERSON = gql`
    query {
        currentUser {
          id
            profile {
                firstName
                lastName

                email
                phoneNumber

                gender
                ethnicity
                birthDate

                communicationPreferences {
                  allowPushNotification
                  allowSMS
                  allowEmail
                }

                campus {
                    featuredImage {
                        uri
                    }
                    name
                }

                photo {
                    uri
                }
                
                address {
                    street1
                    city
                    state
                    postalCode
                }
            }      
        }
    }
`