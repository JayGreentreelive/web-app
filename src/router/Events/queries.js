import gql from 'graphql-tag'

export const GET_EVENT = gql`
    query getEvent($title: String!) {
        getEventContentByTitle(title:$title) {
            id
            title
            summary
            htmlContent
            
            coverImage {
                name
                sources {
                    uri
                }
            }
            
            tags
            
            callsToAction {
                call
                action
            }

            events {
                start
                end
                campuses {
                    name
                }
                location
            }
        }
    }
`

export const GET_EVENTS = gql`
    query getEvents {
        allEvents {
            id
            title
            startDate
            endDate

            events {
                start
                end
            }
        }
    }
`
