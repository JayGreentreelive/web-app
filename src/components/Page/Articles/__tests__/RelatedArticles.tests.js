import React from 'react'
import { MockedProvider } from '@apollo/react-testing'
import { act, render } from '@testing-library/react'
import wait from 'waait'

import { GET_RELATED_ARTICLES, GET_TOP_THREE_ARTICLES } from '../queries'
import { RelatedArticles } from '..'

const createArticle = (id) => ({
    "id": `ArticleContentItem:${id}`,
    "title": `Article ${id}`,
    "summary": "READ THIS GUYS!",
    "images": [
        {
            "sources": [
                {
                    "uri": "https://dev-rock.christfellowship.church/GetImage.ashx?guid=54ef1562-4e7b-4012-9630-115e056554e5"
                }
            ]
        }
    ],
})

const RELATED_ARTICLES_MOCKS = [
    {
        request: {
            query: GET_RELATED_ARTICLES,
            variables: { id: "MainArticle" }
        },
        result: {
            data: {
                node: {
                    siblingContentItemsConnection: {
                        edges: [
                            { node: createArticle(1) },
                            { node: createArticle(2) },
                            { node: createArticle(3) },
                        ]
                    }
                }
            },
        },
    },
]

const TOP_THREE_ARTICLES_MOCKS = [
    {
        request: {
            query: GET_RELATED_ARTICLES,
            variables: { id: "MainArticle" }
        },
        result: {
            data: {
                node: {
                    siblingContentItemsConnection: {
                        edges: []
                    }
                }
            },
        },
    },
    {
        request: {
            query: GET_TOP_THREE_ARTICLES,
        },
        result: {
            data: {
                getArticles: [
                    createArticle(1),
                    createArticle(2),
                    createArticle(3),
                ]
            },
        },
    },
]

const RELATED_ARTICLES_ERROR_MOCKS = [
    {
        request: {
            query: GET_RELATED_ARTICLES,
        },
        error: new Error("Error loading related articles")
    }
]

const RELATED_ARTICLES_ERROR_MOCKS = [
    {
        request: {
            query: GET_RELATED_ARTICLES,
        },
        error: new Error("Error loading related articles")
    },
    {
        request: {
            query: GET_TOP_THREE_ARTICLES,
        },
        error: new Error("Error loading top articles")
    },
]

let component = null

describe("ArticleList", () => {
    it("renders without crashing", () => {
        act(() => {
            render(
                <MockedProvider mocks={RELATED_ARTICLES_MOCKS} addTypename={false}>
                    <RelatedArticles id="MainArticle" />
                </MockedProvider>
            )
        })
    })

    it("renders the loading state", () => {
        act(() => {
            component = render(
                <MockedProvider mocks={[]}>
                    <RelatedArticles id="MainArticle" />
                </MockedProvider>
            )
        })

        const { container } = component
        expect(container).toMatchSnapshot()
    })

    it("renders the error state", async () => {
        act(() => {
            component = render(
                <MockedProvider mocks={RELATED_ARTICLES_ERROR_MOCKS}>
                    <RelatedArticles id="MainArticle" />
                </MockedProvider>
            )
        })

        await wait(0) // waits for response

        const { container } = component
        expect(container).toMatchSnapshot()
    })

    it("renders first three related articles", async () => {
        act(() => {
            component = render(
                <MockedProvider mocks={RELATED_ARTICLES_MOCKS} addTypename={false}>
                    <RelatedArticles id="MainArticle" />
                </MockedProvider>
            )
        })

        await wait(0) // waits for response

        const { container } = component
        expect(container).toMatchSnapshot()
    })

    // Top Three Articles
    it("renders the first three articles when no related articles are found", () => {
        act(() => {
            component = render(
                <MockedProvider mocks={TOP_THREE_ARTICLES_MOCKS}>
                    <RelatedArticles id="MainArticle" />
                </MockedProvider>
            )
        })

        const { container } = component
        expect(container).toMatchSnapshot()
    })
})