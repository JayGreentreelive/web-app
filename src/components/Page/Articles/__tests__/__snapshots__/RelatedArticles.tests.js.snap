// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedArticles renders first three related articles 1`] = `
<div>
  <div
    class="container-fluid max-width-1100 my-6 px-3"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="text-dark align-self-start"
        >
          Related Articles
        </h3>
      </div>
      <div
        class="col text-right"
      >
        <a
          class="text-dark align-self-end"
          href="/articles"
        >
          See More
        </a>
      </div>
    </div>
    <div
      class="row"
    >
      <a
        class="col-12 col-md-4 py-1 pr-1 text-white"
        href="/articles/article-1"
      >
        <div
          class="embed-responsive embed-responsive-1by1 rounded "
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="w-100 h-100 p-3 d-flex flex-row align-items-end"
              style="z-index: 2;"
            >
              <div>
                <h3>
                  Article 1
                </h3>
              </div>
            </div>
            <div
              class="p-absolute w-100 h-100"
              style="top: 0px; left: 0px; overflow: hidden; background: rgba(0, 0, 0, 0.4); z-index: 0;"
            />
          </div>
        </div>
      </a>
      <a
        class="col-12 col-md-4 py-1 pr-1 text-white"
        href="/articles/article-1"
      >
        <div
          class="embed-responsive embed-responsive-1by1 rounded "
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="w-100 h-100 p-3 d-flex flex-row align-items-end"
              style="z-index: 2;"
            >
              <div>
                <h3>
                  Article 1
                </h3>
              </div>
            </div>
            <div
              class="p-absolute w-100 h-100"
              style="top: 0px; left: 0px; overflow: hidden; background: rgba(0, 0, 0, 0.4); z-index: 0;"
            />
          </div>
        </div>
      </a>
      <a
        class="col-12 col-md-4 py-1 pl-1 text-white"
        href="/articles/article-1"
      >
        <div
          class="embed-responsive embed-responsive-1by1 rounded "
        >
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="spinner-border text-primary"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </div>
          <div
            class="fill d-flex justify-content-center align-items-center"
          >
            <div
              class="w-100 h-100 p-3 d-flex flex-row align-items-end"
              style="z-index: 2;"
            >
              <div>
                <h3>
                  Article 1
                </h3>
              </div>
            </div>
            <div
              class="p-absolute w-100 h-100"
              style="top: 0px; left: 0px; overflow: hidden; background: rgba(0, 0, 0, 0.4); z-index: 0;"
            />
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`RelatedArticles renders the error state 1`] = `<div />`;

exports[`RelatedArticles renders the first three articles when no related articles are found 1`] = `<div />`;

exports[`RelatedArticles renders the loading state 1`] = `
<div>
  <div
    class="fill d-flex justify-content-center align-items-center"
  >
    <div
      class="spinner-border text-primary"
      role="status"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
